fdbserver/BlobManager.actor.cpp

```diff
@@ -701,27 +701,14 @@ ACTOR Future<Void> maybeSplitRange(BlobManagerData* bmData, UID currentWorkerId,
 	return Void();
 }
 
-ACTOR Future<Void> monitorBlobWorker(BlobManagerData* bmData, BlobWorkerInterface bwInterf) {
-	try {
-		state Future<Void> waitFailure = waitFailureClient(bwInterf.waitFailure, SERVER_KNOBS->BLOB_WORKER_TIMEOUT);
-		state ReplyPromiseStream<GranuleStatusReply> statusStream =
-		    bwInterf.granuleStatusStreamRequest.getReplyStream(GranuleStatusStreamRequest(bmData->epoch));
-		state KeyRangeMap<std::pair<int64_t, int64_t>> lastSeenSeqno;
-
-		loop choose {
-			when(wait(waitFailure)) {
-				// FIXME: actually handle this!!
-				if (BM_DEBUG) {
-					printf("BM %lld detected BW %s is dead\n", bmData->epoch, bwInterf.id().toString().c_str());
-				}
-				// get all of its ranges
-				// send revoke request to get back all its ranges
-				// send halt (look at rangeMover)
-				// send all its ranges to assignranges stream
-				return Void();
-			}
-			when(GranuleStatusReply _rep = waitNext(statusStream.getFuture())) {
-				GranuleStatusReply rep = _rep;
+ACTOR Future<Void> monitorBlobWorkerStatus(BlobManagerData* bmData, BlobWorkerInterface bwInterf) {
+	state KeyRangeMap<std::pair<int64_t, int64_t>> lastSeenSeqno;
+	loop {
```

Comment:
nit: could this be simplified to 
```loop { try { ... } catch { ... } }``` instead of 
```loop { try { loop { ... } } catch { ... } }```?

Reply:
Ah yeah so the reason for the two loops is the outer loop is handling retries for one instance of the stream (constructing and reading from it)
the inner loop is for reading from a stream once constructed
The idea being that if you get an error from either constructing or reading from a stream, you want to reconstruct the stream and start over, essentially.
I can add a comment that clarifies this though.

Reply:
Ah okay, I missed that. thanks for adding the comment!